src/drive/drive-gateway.ts

```diff
@@ -0,0 +1,329 @@
+    // }
+    // const rParams = url.getParamsResult("auth", "name");
+    // if (rParams.isErr()) {
+    //   return this.logger.Error().Url(url).Err(rParams).Msg("Put Error").ResultError();
+    // }
+    // const { auth } = rParams.Ok();
+    // const { name } = rParams.Ok();
+
+    // const fileId = await search(this.logger, name, auth);
+    // if (fileId.Err()) {
+    //   return fileId;
+    // }
+    // const fileMetadata = await get(this.logger, fileId.Ok(), "fileMetaData", auth);
+    // const fileData = new Blob([new Uint16Array(0)], { type: "application/json" });
+    //   const done = await update(this.logger, fileId, fileMetadata, fileData, auth);
+    //   if (!done) {
+    //     return this.logger.Error().Url(url).Msg(`failed to update ${pathPart}`).ResultError();
+    //   }
+    //   return Result.Ok(undefined);
+    // } else {
+    //   return this.logger.Error().Url(url).Err(rParams).Msg("Database not found").ResultError();
+    // }
+  }
+
+  async start(uri: URI): Promise<Result<URI>> {
+    this.logger.Debug().Str("url", uri.toString()).Msg("start");
+    const ret = uri.build().defParam("version", "v0.1-gdrive").URI();
+    return Result.Ok(ret);
+  }
+
+  async close(): Promise<bs.VoidResult> {
+    return Result.Ok(undefined);
+  }
+
+  async put(url: URI, body: Uint8Array): Promise<bs.VoidResult> {
+    const rParams = url.getParamsResult("auth", "name", "store");
+    if (rParams.isErr()) {
+      return this.logger.Error().Url(url).Err(rParams).Msg("Put Error").ResultError();
+    }
+    const { auth, store } = rParams.Ok();
+    let { name } = rParams.Ok();
+    const index = url.getParam("index");
+    if ('undefined' !== typeof index) {
+      name += `-${index}`;
+    }
+    const fileId = await this.#search(name, auth, store);
+    if (fileId.isErr()) {
+      if (isNotFoundError(fileId.Err())) {
+        const fileData = new Blob([body], { type: "text/plain" });
+
+        const done = await this.#insert(name, body, auth, store);
+        if (done.isErr()) {
+          return done;
+        }
+        return Result.Ok(undefined);
+      }
+    }
+    const fileData = new Blob([body], { type: "application/octet-stream" });
+
+    const done = await this.#update(fileId.Ok(), fileData, auth, store);
+    if (done.isErr()) {
+      return done;
+    }
+    return Result.Ok(undefined);
+  }
+
+  async #delete(fileId: string, auth: string): Promise<Result<unknown>> {
+    const url = this.params.driveURL;
+    const headers = {
+      Authorization: `Bearer ${auth}`,
+    };
+    try {
+      const response = await fetch(url + fileId, {
+        method: "DELETE",
+        headers: headers,
+      });
+      return await response.json();
+    } catch (err) {
+      return this.logger.Error().Url(url).Any("init", auth).Err(err).Msg("Could not delete").ResultError();
+    }
+  }
+
+  async #get(fileId: string, auth: string): Promise<Result<Uint8Array>> {
+    let response;
+    let headers;
```

Comment:
pls use const there are very rare situation where let is need and our lint will not accept this.